edweiss-app/app/(app)/lectures/slides/index.tsx

```diff
@@ -99,35 +103,52 @@ const LectureScreen: ApplicationRoute = () => {
         setUri(url);
     }
 
-    function updateUI() {
+    // Function for updating the UI display values when switching through orientations an fullscreen modes
+    function updateUI(orientation: ScreenOrientation.Orientation) {
+        const bool = ((orientation == ScreenOrientation.Orientation.LANDSCAPE_LEFT || orientation == ScreenOrientation.Orientation.LANDSCAPE_RIGHT))
+        setIsLandscape(bool);
         if (isFullscreen) {
             setWidthPercent(["100%", "0%"]);
             setHeightPercent(["100%", "0%"]);
-        } else if (isLandscape) {
+        } else if (bool) {
             setHeightPercent(["100%", "100%"]);
             setWidthPercent(["60%", "40%"]);
         } else {
             setHeightPercent(["40%", "60%"]);
             setWidthPercent(["100%", "100%"]);
```

Comment:
That great to use it this way. I propose a minor enhancement. Create constant for each state of the if statement and use ternary operator to fix it. but just watch out to do not break the consistency of the code. Here's what I was thinkin of:

`
const W_FULL  = ["100%", "0%"];
const H_FULL  = ["100%", "0%"];
const W_BOOL    = ["60%", "40%"];
const H_BOOL    = ["100%", "100%"];
const W_DEF       = ["100%", "100%"];
const H_DEF       = ["40%", "60%"];

setWidthPercent((isFullScreen) ? W_FULL : ((bool) ? W_BOOL : W_DEF));
setHeightPercent((isFullScreen) ? H_FULL : ((bool) ? H_BOOL : H_DEF));
`
After all this is a suggestion do as you please, focus on functionality than optimization for this milestone.

Reply:
Yes this was actually how I was doing it previously but the sonarcloud complained about too many uses of ternary operators so I changed it into this format. It is not very clean I know once I have more time I will try to rearrange this